.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,53 @@
+name: Docker Publish
+
+on:
+  workflow_run:
+    workflows : ["Docker build"]
+    branches: ["master"]
+    types:
+      - completed
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  docker:
+    runs-on: ubuntu-20.04
+    # Run if this is a push on master and Docker Build succeeded
+    # or if a tag was pushed
+    if: ${{ github.event.workflow_run.conclusion == 'success' ||  github.ref_name != 'master' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Get short sha
+        id: sha
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: ${{ matrix.platform }}
+          push: true
+          # If this is a push on master, publish with short commit sha
+          # else use the ref name, which has to be the tag in this case.
+          tags: >
+            ${{ secrets.DOCKERHUB_USERNAME }}/qlever:${{ github.ref_name == 'master' && steps.sha.outputs.sha_short || github.ref_name }},
+            ${{ secrets.DOCKERHUB_USERNAME }}/qlever:latest
```

Comment:
Add a comment `We have to explicitly add the "qlever:latest" tag for it to work correctly, see e.g. https://stackoverflow.com/questions/27643017/do-i-need-to-manually-tag-latest-when-pushing-to-docker-public-repository`

(We are not docker experts, so some comments about edge cases are always appreciated:)